nemo_curator/modules/distributed_data_classifier.py

```
@@ -96,6 +150,126 @@ def _filter_documents(
+        op.Predictor(
+            model,
+            sorted_data_loader=True,
+            batch_size=batch_size,
+            pred_output_col=prob_internal_col,
+        ),
+        repartition=df.npartitions,
+        keep_cols=columns_to_keep_list,
+    )
+    df = classifier_pipe(df)
+    # TODO: Make crossfit handle this cleanly
+    # to prevent the labeler from dropping the prob_internal_col
+    # and combine it into a single step
+    labeling_pipe = op.Sequential(
+        op.Labeler(labels, cols=[prob_internal_col]),
+        keep_cols=columns_to_keep_list + [prob_internal_col],
+    )
+    df = labeling_pipe(df)
+    if keep_prob:
+        df = df.rename(
+            columns={pred_internal_col: label_col, prob_internal_col: prob_col}
+        )
+    else:
+        df = df.rename(columns={pred_internal_col: label_col})
+        df = df.drop(columns=[prob_internal_col])
+
+    return df
+
+
+class DomainModel(HFModel):
+    def __init__(self, config, out_dim=None, model_path=None, autocast=False):
+        self.config = config
+        self.out_dim = out_dim
+        self.model_path = model_path
+        self.autocast = autocast
+        super().__init__(self.config.model)
+
+    def load_model(self, device="cuda"):
+        model = CustomModel(
+            self.config,
+            out_dim=self.out_dim,
+            config_path=None,
+            pretrained=True,
+            autocast=self.autocast,
+        )
+        model = model.to(device)
+        if os.path.exists(self.model_path):
+            sd = torch.load(os.path.join(self.model_path), map_location="cpu")
+            sd = {k[7:] if k.startswith("module.") else k: sd[k] for k in sd.keys()}
+            if version.parse(TRANSFORMERS_VERSION) >= version.parse("4.31.0"):
+                sd.pop("model.embeddings.position_ids", None)
+            model.load_state_dict(sd, strict=True)
+        return model.eval()
```

If `model_path` doesn't exist, does just the base model get returned? We should probably throw an error instead.

---

Resolved